core/service/src/engine/threshold/service/public_decryptor.rs

```diff
@@ -333,17 +333,16 @@ impl<
                     "Public decryption request with ID {req_id} already exists"
                 )));
             }
-
-            if !ok_or_tonic_abort(
-                self.crypto_storage.threshold_fhe_keys_exists(&key_id).await,
-                "Could not check threshold fhe key existance".to_string(),
-            )? {
-                return Err(Status::not_found(format!(
-                    "Threshold FHE keys with key ID {key_id} not found"
-                )));
-            }
         }
 
+        self.crypto_storage
+            .refresh_threshold_fhe_keys(&key_id)
```

Comment:
Why is it actually needed to execute the refresh? I thought we always load the data into the cache at boot and after constructing keys?

Reply:
This has been in both decrypt calls for a long time. This PR is just intended as a hotfix, so I didn't want to dig too deep into the logic. But I agree we should assess if this makes sense. 
However with what's implemented here now, we do use the refresh to check if the key exists (in cache, which if more efficient than locking both storages).

Reply:
Yes, totally agree! Just want to make to make sure that we are not missing another change that could make everything even more efficient :)